File: www/html/cgi-bin/system_stat.py
#!/usr/bin/env python3
import os
import sys
from datetime import datetime

# CGI requires standard HTTP headers separated by a blank line
print("Content-Type: text/plain")
print("X-Powered-By: Sentinel-CGI")
print("")

# Output dynamic system status
print("--- SYSTEM DIAGNOSTICS ---")
print(f"Timestamp : {datetime.now().strftime('%Y-%m-%d %H:%M:%S')}")
print(f"Method    : {os.environ.get('REQUEST_METHOD', 'UNKNOWN')}")
print(f"Protocol  : {os.environ.get('SERVER_PROTOCOL', 'UNKNOWN')}")
print(f"Query     : {os.environ.get('QUERY_STRING', 'NONE')}")

print("\n--- CLIENT HEADERS ---")
# Extract and print HTTP_ variables passed by your C++ engine
for key, value in sorted(os.environ.items()):
    if key.startswith("HTTP_"):
        header_name = key[5:].replace("_", "-").title()
        print(f"{header_name}: {value}")

# If it's a POST request, read stdin
if os.environ.get("REQUEST_METHOD") == "POST":
    print("\n--- POST BODY ---")
    body = sys.stdin.read(1024)  # Read up to 1KB for safety
    print(body if body else "<Empty>")

File: www/html/cgi-bin/system_stat.php
#!/usr/bin/php-cgi
<?php
// Enforce plain text output for the Sentinel OS terminal
header("Content-Type: text/plain");
header("X-Powered-By: Sentinel-PHP-Engine");

echo "--- PHP SYSTEM DIAGNOSTICS ---\n";
echo "Timestamp : " . date('Y-m-d H:i:s') . "\n";
echo "PHP Core  : " . phpversion() . "\n";
echo "Method    : " . ($_SERVER['REQUEST_METHOD'] ?? 'UNKNOWN') . "\n";
echo "Query     : " . ($_SERVER['QUERY_STRING'] ?? 'NONE') . "\n";

echo "\n--- CLIENT HEADERS ---\n";
// Extract and format HTTP headers provided by the C++ engine
foreach ($_SERVER as $key => $value) {
    if (strpos($key, 'HTTP_') === 0) {
        $headerName = str_replace(' ', '-', ucwords(strtolower(str_replace('_', ' ', substr($key, 5)))));
        echo "$headerName: $value\n";
    }
}

// Read raw body for POST requests
if (($_SERVER['REQUEST_METHOD'] ?? '') === 'POST') {
    echo "\n--- POST BODY ---\n";
    $body = file_get_contents("php://input");
    echo $body ? $body : "<Empty>";
}
?>